textdb/textdb-storage/src/main/java/edu/uci/ics/textdb/storage/relation/RelationManager.java

```diff
@@ -217,6 +217,7 @@ private void insertTupleToDirectory(IDataStore dataStore, Analyzer luceneAnalyze
         
         DataWriter dataWriter = new DataWriter(new DataStore(tableDirectory, tableSchema), luceneAnalyzer);
         dataWriter.insertTuple(tuple);
+        dataWriter.close();
```

Comment:
Yes this is correct. The problem now is RelationManager doesn't provide the abstraction of inserting multiple tuples. We need to find some other way to add similar `open()` and `close()` to RelationManager. One way is to go back to the `DataWriter` and `DataReader` level. Another way I can think of is to add similar `open()` and `close()` in RelationManager. I'm also thinking if there are better ways to do it.